### PR TITLE
feat(protocol): Add stub metric summaries to spans and the event

### PR DESCRIPTION
**Pull request by mitsuhiko**
To make our live easier add `_metrics_summary` temporarily to the protocol to enable iteration on span correlations.

Refs https://github.com/getsentry/sentry-python/pull/2522

### CODE REVIEW

File: relay-event-schema/src/protocol/event.rs [hunk 1]
```diff
@@ -491,6 +491,13 @@ pub struct Event {
     #[metastructure(omit_from_schema)]
     pub _metrics: Annotated<Metrics>,
 
+    /// Temporary protocol support for metric summaries.
+    ///
+    /// This shall move to a stable location once we have stabilized the
+    /// interface.  This is intentionally not typed today.
+    #[metastructure(omit_from_schema)]
+    pub _metrics_summary: Annotated<Object<Value>>,
```
Comment (jjbayer) [inline relay-event-schema/src/protocol/event.rs]: We could even make it `Annotated<Value>` to get maximum flexibility.

File: relay-event-schema/src/protocol/span.rs [hunk 2]
```diff
@@ -76,6 +76,12 @@ pub struct Span {
     /// Timestamp when the span has been received by Sentry.
     pub received: Annotated<Timestamp>,
 
+    /// Temporary protocol support for metric summaries.
+    ///
+    /// This shall move to a stable location once we have stabilized the
+    /// interface.  This is intentionally not typed today.
+    pub _metrics_summary: Annotated<Object<Value>>,
+
```
Comment (jjbayer) [inline relay-event-schema/src/protocol/span.rs]: Properties on spans are retained anyway (see `additional_properties` with `retain=true` below), so this addition is probably not even necessary.

**Review by jjbayer — APPROVED** [review]
See comment about `additional_properties`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,6 +18,7 @@
 - Support comparison operators (`>`, `>=`, `<`, `<=`) for strings in dynamic sampling and metric extraction rules. Previously, these comparisons were only possible on numbers. ([#2730](https://github.com/getsentry/relay/pull/2730))
 - Postpone processing till the global config is available. ([#2697](https://github.com/getsentry/relay/pull/2697))
 - Skip running `NormalizeProcessor` on renormalization. ([#2744](https://github.com/getsentry/relay/pull/2744))
+- Add stub metric summaries. ([#2757](https://github.com/getsentry/relay/pull/2757))
```
Comment (olksdr) [inline CHANGELOG.md]: Please, move changelog entry to `Unreleased` section.

File: relay-event-schema/src/protocol/span.rs (hunk 2 above)
Comment (jan-auer) [reply relay-event-schema/src/protocol/span.rs]: Good catch, I missed that when talking to Armin!

**Pull request merged**